One of the test failures looks real
```
FAIL: corehq.apps.app_manager.xpath_validator.tests:XpathValidatorTest.test_bad_hashtag
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/xpath_validator/tests.py", line 60, in test_bad_hashtag
    message=b"hashtag is not a valid # expression\n"))
AssertionError: Xpath[21 chars]_valid=False, message='hashtag is not a valid # expression\n') != Xpath[21 chars]_valid=False, message=b'hashtag is not a valid # expression\n')
```

--- a/corehq/apps/app_manager/xpath_validator/tests.py
+++ b/corehq/apps/app_manager/xpath_validator/tests.py
@@ -17,7 +17,7 @@ class XpathValidatorTest(SimpleTestCase):
         self.assertEqual(
             validate_xpath('data node'),
             XpathValidationResponse(is_valid=False, message=
-b"""Lexical error on line 1. Unrecognized text.
+"""Lexical error on line 1. Unrecognized text.
 data node
 -----^
 """))
@@ -43,7 +43,7 @@ data node
             validate_xpath(
                 "if(count(instance('commcaresession')/session/user/data/commcare_location_id) &gt; 0, instance('commcaresession')/session/user/data/commcare_location_id, /data/meta/userID)"),
             XpathValidationResponse(is_valid=False, message=
-b"""Lexical error on line 1. Unrecognized text.
+"""Lexical error on line 1. Unrecognized text.
 ...mmcare_location_id) &gt; 0, instance('co
 -----------------------^
 """))
@@ -57,7 +57,7 @@ b"""Lexical error on line 1. Unrecognized text.
         self.assertEqual(
             validate_xpath('#hashtag'),
             XpathValidationResponse(is_valid=False,
-                                    message=b"hashtag is not a valid # expression\n"))
+                                    message="hashtag is not a valid # expression\n"))
 
     def test_case_hashtag(self):
         self.assertEqual(
@@ -68,4 +68,4 @@ b"""Lexical error on line 1. Unrecognized text.
         self.assertEqual(
             validate_xpath('#case', allow_case_hashtags=False),
             XpathValidationResponse(is_valid=False,
-                                    message=b"case is not a valid # expression\n"))
+                                    message="case is not a valid # expression\n"))

--- a/corehq/apps/app_manager/xpath_validator/wrapper.py
+++ b/corehq/apps/app_manager/xpath_validator/wrapper.py
@@ -5,7 +5,6 @@ from corehq.apps.app_manager.xpath_validator.config import get_xpath_validator_p
 from corehq.apps.app_manager.xpath_validator.exceptions import XpathValidationError
 from dimagi.utils.subprocess_manager import subprocess_context
 import six
-import re
 
 XpathValidationResponse = namedtuple('XpathValidationResponse', ['is_valid', 'message'])
 
@@ -23,7 +22,7 @@ def validate_xpath(xpath, allow_case_hashtags=False):
             stdout, stderr = p.communicate(xpath)
         else:
             p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            stdout, stderr = p.communicate(re.sub(r'\s+', ' ', xpath).encode('utf-8'))
+            stdout, stderr = p.communicate(xpath.encode('utf-8'))
         exit_code = p.wait()
     if exit_code == 0:
         return XpathValidationResponse(is_valid=True, message=None)

